index.js

```diff
@@ -42,14 +44,30 @@ module.exports = function requireAll(options) {
         filter: filter,
         excludeDirs: excludeDirs,
         map: map,
-        resolve: resolve
+        resolve: resolve,
+        merge: merge
       });
 
     } else {
       var name = filterFile(file);
       if (!name) return;
 
-      modules[map(name, filepath)] = resolve(require(filepath));
+      var module = resolve(require(filepath));
+
+      if (merge) {
+        // console.log('module and is object', module, isObject(module))
+      if (isObject(module)) {
+        for (let key in module) {
+          modules[key] = module[key];
```

Comment:
let this as is for reasons I explained.  the map function was really intended to allow changing the filename to something else as the prop not to map propskeys within the actual module